core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -303,11 +303,15 @@ class AdminManager(val config: KafkaConfig,
 
           case ResourceType.TOPIC =>
             val topic = resource.name
-            Topic.validate(topic)
-            // Consider optimizing this by caching the configs or retrieving them from the `Log` when possible
-            val topicProps = adminZkClient.fetchEntityConfig(ConfigType.Topic, topic)
-            val logConfig = LogConfig.fromProps(KafkaServer.copyKafkaConfigToLog(config), topicProps)
-            createResponseConfig(allConfigs(logConfig), createTopicConfigEntry(logConfig, topicProps, includeSynonyms))
+            if (metadataCache.contains(topic)) {
+              Topic.validate(topic)
+              // Consider optimizing this by caching the configs or retrieving them from the `Log` when possible
+              val topicProps = adminZkClient.fetchEntityConfig(ConfigType.Topic, topic)
```

Comment:
Shouldn't `fetchEntityConfig` indicate that the topic doesn't exist?

Reply:
@ijuma  Thanks for the review.  fetchEntityConfig return empty props for non-existing entities. We can add a empty check. But i thought using metadataCache is consistent with other methods in KafkaAPIs. Let me you know your preference.

Reply:
I think `fetchEntityConfigs` should return an `Option` and `None` for the case where the node doesn't exist.

Reply:
Yes, fetchEntityConfigs can return Optional result. I can update this as part of separate PR.

Reply:
I'm suggesting that it's a better approach than the one in the current PR.

Reply:
Oh sorry, I had not seen your edit. That is weird that we do not delete the znode. Is there a reason for that? We should file a JIRA. In that case, it does seem like we have to rely on the metadata cache.

Reply:
@ijuma  Your suggestion works. We are deleting config znode, when we delete topic. We are not deleting znode, when we don't have any custom configs. 
My only concern is, Do we need to depend on config znode for topic existence check?  Someone may create topic with custom  zk script , which may not create config znode while creating topic.